### ISSUE TITLE
On the fly editing

### ISSUE THREAD

**Issue by jappeace** [jappeace/cut-the-crap#33]
Rather then editing a video before watching it, it'd be nicer to directly integrate this project into a player.

Technically I'm not sure how to do this, some possibilities:

+ Spit out chunks of video to feed into a player
+ directly hack the editing into the encoder/decoder.

**Comment by lijovklm** [comment]
this looks similar to the [skip silence](https://github.com/vantezzen/skip-silence) chrome addon

**Comment by jappeace** [comment]
@lijovklm that's pretty cool, it goes the other way though, cutting out uninetersting parts for videos that have already been published, whereas this project allows the publisher to cut out unintersting parts.
Particularly for the youtube case this is usefull, where that other project mentions:

>  On some websites (e.g. YouTube) this "pre-buffering" is not possible though, as their media elements can not be cloned.

**Comment by lijovklm** [comment]
@jappeace that sounds interesting, Btw I was never able to try this project. I got here from the jumpcutter thread. is it possible to install this in a windows pc? If needed just let me know, I will open this as an issue.

**Comment by jappeace** [comment]
well, I'm trying to host this as a webserver: https://videocut.org/

It's not quite working yet, turns out uploading large files isn't that easy. But I should have something useable probably by next week (I'll hack on it again this weekend in fact). I'm doing that because I want non-engineers to use this as well. 

I'd be happy to accept any PR's that add windows support to this project as well. The core of the program is ffmpeg, so it should be supported: https://ffmpeg.org/download.html
It may be simply an issue of getting it to work with stack (because I know stack builds haskell on linux).

There is also a debian package which may be installed into WSL if you don't care about native support.

**Comment by lijovklm** [comment]
Tried the webserver with a small file of 8mb in size, but it failed.
Input file was uploading for some time, and then the page crashed.

I would more like this as a desktop application or CLI application rather than uploading files to a web server.

**Comment by lijovklm** [comment]
I am completely unfamiliar with the Haskell world, but let me take a look at the code.

**Comment by jappeace** [comment]
@lijovklm yeah, like I mentioned, it's a work in progress. I'll ping you once the webserver reliably working. Don't get dragged down into setting up that windows build, haskell's build tooling is notoriously confusing. But if you have time, you want to look into stack. Ask me any questions either if you get stuck.

**Comment by lijovklm** [comment]
Sure thing. how do I contact you?  You in telegram?

**Comment by jappeace** [comment]
@lijovklm preferably on discord: https://discord.gg/Hp4agqy
